gtk4/src/subclass/editable.rs

```diff
@@ -0,0 +1,154 @@
+    }
+}
+
+unsafe extern "C" fn editable_insert_text<T: EditableImpl>(
+    editable: *mut ffi::GtkEditable,
+    text: *const c_char,
+    length: c_int,
+    position: *mut c_int,
+) {
+    let instance = &*(editable as *mut T::Instance);
+    let imp = instance.get_impl();
+
+    imp.insert_text(
+        from_glib_borrow::<_, Editable>(editable).unsafe_cast_ref(),
+        &GString::from_glib_borrow(text),
+        length,
+        &mut *position,
+    )
+}
+
+unsafe extern "C" fn editable_delete_text<T: EditableImpl>(
+    editable: *mut ffi::GtkEditable,
+    start_position: c_int,
+    end_position: c_int,
+) {
+    let instance = &*(editable as *mut T::Instance);
+    let imp = instance.get_impl();
+
+    imp.delete_text(
+        from_glib_borrow::<_, Editable>(editable).unsafe_cast_ref(),
+        start_position,
+        end_position,
+    )
+}
+
+unsafe extern "C" fn editable_changed<T: EditableImpl>(editable: *mut ffi::GtkEditable) {
+    let instance = &*(editable as *mut T::Instance);
+    let imp = instance.get_impl();
+
+    imp.changed(from_glib_borrow::<_, Editable>(editable).unsafe_cast_ref())
+}
+
+unsafe extern "C" fn editable_get_text<T: EditableImpl>(
+    editable: *mut ffi::GtkEditable,
+) -> *const c_char {
+    let instance = &*(editable as *mut T::Instance);
+    let imp = instance.get_impl();
+
+    imp.get_text(from_glib_borrow::<_, Editable>(editable).unsafe_cast_ref())
+        .to_glib_full()
+}
+
+unsafe extern "C" fn editable_do_insert_text<T: EditableImpl>(
+    editable: *mut ffi::GtkEditable,
+    text: *const c_char,
+    length: i32,
+    position: *mut i32,
+) {
+    let instance = &*(editable as *mut T::Instance);
+    let imp = instance.get_impl();
+
+    imp.do_insert_text(
+        from_glib_borrow::<_, Editable>(editable).unsafe_cast_ref(),
+        &GString::from_glib_borrow(text),
+        length,
+        &mut *position,
+    )
+}
+
+unsafe extern "C" fn editable_do_delete_text<T: EditableImpl>(
+    editable: *mut ffi::GtkEditable,
+    start_position: i32,
+    end_position: i32,
+) {
+    let instance = &*(editable as *mut T::Instance);
+    let imp = instance.get_impl();
+
+    imp.do_delete_text(
+        from_glib_borrow::<_, Editable>(editable).unsafe_cast_ref(),
+        start_position,
+        end_position,
+    )
+}
+
+unsafe extern "C" fn editable_get_selection_bounds<T: EditableImpl>(
+    editable: *mut ffi::GtkEditable,
+    start_position: *mut i32,
+    end_position: *mut i32,
+) -> glib::ffi::gboolean {
+    let instance = &*(editable as *mut T::Instance);
+    let imp = instance.get_impl();
+
+    if let Some((start_pos, end_pos)) =
+        imp.get_selection_bounds(from_glib_borrow::<_, Editable>(editable).unsafe_cast_ref())
+    {
+        if !start_position.is_null() {
+            *start_position = start_pos;
+        }
+
+        if !end_position.is_null() {
+            *end_position = end_pos;
+        }
+        true.to_glib()
+    } else {
+        false.to_glib()
```

Comment:
You might want to set the two values to `0` here as the docs at least say that the positions should have the same value when returning `FALSE`